Applications/WebPortal/WebApp/tests/entitlements_page_tester/source/index.tsx

```diff
@@ -1,61 +1,217 @@
-const app = new Beam.Map<Nexus.EntitlementKey, Nexus.MarketDataTypeSet>();
-app.set(ekey1, dataset1);
-app.set(ekey2, dataset2);
-
-const entitlementEntry1 = new Nexus.EntitlementDatabase.Entry('ASX Total',
-  Nexus.Money.parse('68'), Nexus.DefaultCurrencies.USD, group, app);
-const cEntry1 = currencyDB.fromCode('USD');
-const entitlementEntry2 = new Nexus.EntitlementDatabase.Entry('TSX Venture',
-  Nexus.Money.parse('200'), Nexus.DefaultCurrencies.EUR, group, app);
-const cEntry2 = currencyDB.fromCode('EUR');
-
-const test = (
-  <WebPortal.HBoxLayout height='100%' width='100%'>
-    <WebPortal.Padding size='20%'/>
-    <WebPortal.VBoxLayout height='100%' width='100%'>
-      <WebPortal.Padding size='10%'/>
-      <WebPortal.EntitlementRow
-        isActive
-        currencyEntry={cEntry1}
-        entitlementEntry={entitlementEntry1}
-        displaySize={WebPortal.DisplaySize.SMALL}
-        marketDatabase={marketDB}/>
-      <WebPortal.EntitlementRow
-        isActive
-        currencyEntry={cEntry1}
-        entitlementEntry={entitlementEntry1}
-        displaySize={WebPortal.DisplaySize.MEDIUM}
-        marketDatabase={marketDB}/>
-      <WebPortal.EntitlementRow
-        isActive
-        currencyEntry={cEntry1}
-        entitlementEntry={entitlementEntry1}
-        displaySize={WebPortal.DisplaySize.LARGE}
-        marketDatabase={marketDB}/>
-      <WebPortal.EntitlementRow
-        isActive
-        currencyEntry={cEntry1}
-        entitlementEntry={entitlementEntry1}
-        displaySize={WebPortal.DisplaySize.LARGE}
-        marketDatabase={marketDB}/>
-    </WebPortal.VBoxLayout>
-  </WebPortal.HBoxLayout>);
-
-ReactDOM.render(test,
-  document.getElementById('main'));
+/**  Displays a testing application. */
+interface State {
+  roles: Nexus.AccountRoles;
+  displaySize: WebPortal.DisplaySize;
+  checkedDB: Beam.Set<Beam.DirectoryEntry>;
+  currencyDB: Nexus.CurrencyDatabase;
+  marketDB: Nexus.MarketDatabase;
+  entitlementDB: Nexus.EntitlementDatabase;
+  status: string;
+  displayedStatus: string;
+}
+
+class TestApp extends React.Component<{}, State> {
+  constructor(props: {}) {
+    super(props);
+    this.state = {
+      roles:  new Nexus.AccountRoles(),
+      entitlementDB: new Nexus.EntitlementDatabase(),
+      displaySize: TestApp.getDisplaySize(),
+      checkedDB: new Beam.Set<Beam.DirectoryEntry>(),
+      currencyDB: Nexus.buildDefaultCurrencyDatabase(),
+      marketDB: Nexus.buildDefaultMarketDatabase(),
+      status: '',
+      displayedStatus: ''
+    };
+    this.changeRole = this.changeRole.bind(this);
+    this.onScreenResize = this.onScreenResize.bind(this);
+    this.toggleCheckMark = this.toggleCheckMark.bind(this);
+    this.setup = this.setup.bind(this);
+    this.buildEntitlementDB = this.buildEntitlementDB.bind(this);
+    this.changeStatus = this.changeStatus.bind(this);
+    this.commentsSubmitted = this.commentsSubmitted.bind(this);
+  }
+
+  public render(): JSX.Element {
+    this.setup();
```

Comment:
```render``` methods can not be used to modify a component's state. You will have to put this in a ```componentWillMount```.